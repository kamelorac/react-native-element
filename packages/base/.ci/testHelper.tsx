import React, { JSXElementConstructor } from 'react';
import { View } from 'react-native';
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from '@testing-library/react-native';

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  _themeProp: unknown = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };
  const renderApi = render(children, options);
  const wrapper = renderApi.queryByTestId(wrapperTestID || 'wrapper')!;
  return { wrapper, ...renderApi };
};
