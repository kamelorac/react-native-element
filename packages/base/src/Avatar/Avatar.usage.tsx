import React from 'react';
import { Avatar } from '..';
import { usage, Stack, info, meta } from '@rneui/doc-gen';

info(
  'Avatars are found all over ui design from lists to profile screens.',
  'They are commonly used to represent a user and can contain photos, icons, or even text.'
);

meta({
  name: 'Avatar',
});

usage('Shape', '', () => (
  <Stack row spacing={4}>
    <Avatar
      size={32}
      rounded
      icon={{ name: 'pencil', type: 'font-awesome' }}
      containerStyle={{ backgroundColor: '#9700b9' }}
    />
    <Avatar
      size={32}
      icon={{ name: 'pencil', type: 'font-awesome' }}
      containerStyle={{ backgroundColor: '#9700b9' }}
    />
  </Stack>
));

usage('Sizes', '', () => (
  <Stack row spacing={4}>
    <Avatar
      size={24}
      rounded
      icon={{ name: 'pencil', type: 'font-awesome' }}
      containerStyle={{ backgroundColor: '#9700b9' }}
    />
    <Avatar
      size={32}
      rounded
      icon={{ name: 'pencil', type: 'font-awesome' }}
      containerStyle={{ backgroundColor: '#9700b9' }}
    />
    <Avatar
      size={48}
      rounded
      icon={{ name: 'pencil', type: 'font-awesome' }}
      containerStyle={{ backgroundColor: '#9700b9' }}
    />
  </Stack>
));
usage('Image Avatar', '', () => (
  <Stack row spacing={4}>
    <Avatar
      size={32}
      rounded
      source={{ uri: 'https://randomuser.me/api/portraits/men/36.jpg' }}
    />
    <Avatar
      size={32}
      rounded
      source={{ uri: 'https://randomuser.me/api/portraits/men/35.jpg' }}
    />
  </Stack>
));

usage('Letter Avatar', '', () => (
  <Stack row spacing={4}>
    <Avatar
      size={32}
      rounded
      title="Rd"
      containerStyle={{ backgroundColor: 'blue' }}
    />
    <Avatar
      size={32}
      rounded
      title="AB"
      containerStyle={{ backgroundColor: 'purple' }}
    />
  </Stack>
));
