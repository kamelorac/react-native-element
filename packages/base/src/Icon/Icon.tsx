import React from 'react';
import {
  Platform,
  View,
  StyleSheet,
  ViewStyle,
  StyleProp,
  TextStyle,
  Pressable,
  ColorValue,
} from 'react-native';
import {
  IconButtonProps,
  IconProps as VectorIconProps,
} from 'react-native-vector-icons/Icon';
import Color from 'color';
import getIconType from '../helpers/getIconType';
import getIconStyle from '../helpers/getIconStyle';
import {
  androidRipple,
  defaultTheme,
  InlinePressableProps,
  RneFunctionComponent,
} from '../helpers';

export type IconType =
  | 'material'
  | 'material-community'
  | 'simple-line-icon'
  | 'zocial'
  | 'font-awesome'
  | 'octicon'
  | 'ionicon'
  | 'foundation'
  | 'evilicon'
  | 'entypo'
  | 'antdesign'
  | 'font-awesome-5'
  | string;

export interface IconObject {
  /** Name of icon. */
  name?: string;

  /** Color of icon. */
  color?: string;

  /** Size of icon. */
  size?: number;

  /** Type of icon */
  type?: IconType;

  /** Apply style to the icon using iconStyle. */
  iconStyle?: StyleProp<TextStyle>;
}

export type IconNode = boolean | React.ReactElement<{}> | Partial<IconProps>;

export interface IconProps extends InlinePressableProps, IconButtonProps {
  /** Type of icon set. [Supported sets here](#available-icon-sets). */
  type?: IconType;

  /** Update React Native Component.
   *  @default `Press handlers present then Pressable else View`
   */
  Component?: typeof React.Component;

  /** Reverses color scheme. */
  reverse?: boolean;

  /** Adds box shadow to button. */
  raised?: boolean;

  /** Add styling to container holding icon. */
  containerStyle?: StyleProp<ViewStyle>;

  /** Provide all props from react-native Icon component. */
  iconProps?: VectorIconProps;

  /** Specify reverse icon color. */
  reverseColor?: string;

  /** Disables onPress events. Only works when `onPress` has a handler. */
  disabled?: boolean;

  /** Style for the button when disabled. Only works when `onPress` has a handler. */
  disabledStyle?: StyleProp<ViewStyle>;

  /** Uses the solid font. */
  solid?: boolean;

  /** Uses the brands font (FontAwesome5 only). */
  brand?: boolean;
}

/** Icons are visual indicators usually used to describe action or intent.
 * They are also used for displaying information. */
export const Icon: RneFunctionComponent<IconProps> = ({
  type = 'material',
  name,
  size = 24,
  color: colorProp,
  iconStyle,
  iconProps,
  underlayColor = 'transparent',
  reverse = false,
  raised = false,
  containerStyle,
  reverseColor: reverseColorProp,
  disabled = false,
  disabledStyle,
  onPress,
  onLongPress,
  onPressIn,
  onPressOut,
  Component = onPress || onLongPress || onPressIn || onPressOut
    ? Pressable
    : View,
  solid = false,
  brand = false,
  theme = defaultTheme,
  pressableProps,
  ...rest
}) => {
  const color = colorProp || theme?.colors?.black;
  const reverseColor = reverseColorProp || theme?.colors?.white;
  const IconComponent = getIconType(type);
  const iconSpecificStyle = getIconStyle(type, { solid, brand });

  const getBackgroundColor = React.useMemo(() => {
    if (reverse) {
      return color;
    }
    return raised ? theme?.colors?.white : 'transparent';
  }, [color, raised, reverse, theme?.colors?.white]);

  const buttonStyles = React.useMemo(
    () => ({
      borderRadius: size + 4,
      height: size * 2 + 4,
      width: size * 2 + 4,
    }),
    [size]
  );

  return (
    <View
      style={StyleSheet.flatten([
        !raised && styles.container,
        (reverse || raised) && styles.button,
        (reverse || raised) && buttonStyles,
        raised && styles.raised,
        iconStyle && iconStyle.borderRadius
          ? {
              borderRadius: iconStyle.borderRadius,
            }
          : {},
        containerStyle && containerStyle,
      ])}
      testID="RNE__ICON__CONTAINER"
    >
      <Component
        testID="RNE__ICON__CONTAINER_ACTION"
        {...{
          android_ripple: androidRipple(
            Color(reverse ? color : (underlayColor as string))
              .alpha(0.3)
              .rgb()
              .string()
          ),
          onPress,
          onLongPress,
          onPressIn,
          onPressOut,
          disabled,
          accessibilityRole: 'button',
          ...pressableProps,
          ...rest,
        }}
      >
        <View
          style={StyleSheet.flatten([
            (reverse || raised) && buttonStyles,
            {
              backgroundColor: getBackgroundColor as ColorValue,
              alignItems: 'center',
              justifyContent: 'center',
            },
            disabled && styles.disabled,
            disabled && disabledStyle,
          ])}
          testID="RNE__ICON"
        >
          <IconComponent
            testID="RNE__ICON__Component"
            style={StyleSheet.flatten([
              { backgroundColor: 'transparent' },
              iconStyle && iconStyle,
            ])}
            size={size}
            name={name}
            color={reverse ? reverseColor : color}
            {...iconSpecificStyle}
            {...iconProps}
          />
        </View>
      </Component>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    overflow: 'hidden',
  },
  button: {
    margin: 7,
  },
  raised: {
    ...Platform.select({
      android: {
        elevation: 2,
      },
      default: {
        shadowColor: 'rgba(0,0,0, .4)',
        shadowOffset: { height: 1, width: 1 },
        shadowOpacity: 1,
        shadowRadius: 1,
      },
    }),
  },
  disabled: {
    backgroundColor: '#D1D5D8',
  },
});

Icon.displayName = 'Icon';
