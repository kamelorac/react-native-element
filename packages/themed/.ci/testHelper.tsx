import React, { JSXElementConstructor } from 'react';
import { View } from 'react-native';
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from '@testing-library/react-native';
import {
  ThemeProvider,
  FullTheme,
  createTheme,
  CreateThemeOptions,
} from '../src/config';

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  themeProp: Partial<CreateThemeOptions> = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };
  const renderApi = render(
    <ThemeProvider theme={createTheme(themeProp)}>{children}</ThemeProvider>,
    options
  );
  const wrapper = renderApi.queryByTestId(wrapperTestID || 'wrapper')!;
  return { wrapper, ...renderApi };
};
