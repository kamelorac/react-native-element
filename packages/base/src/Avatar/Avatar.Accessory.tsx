import React from 'react';
import {
  Pressable,
  View,
  Platform,
  StyleSheet,
  StyleProp,
  ViewStyle,
  ColorValue,
} from 'react-native';
import { ImageProps, Image } from '../Image';
import { IconProps, Icon } from '../Icon';
import {
  androidRipple,
  InlinePressableProps,
  RneFunctionComponent,
} from '../helpers';

export type AccessoryProps = Partial<IconProps> &
  Partial<ImageProps> &
  InlinePressableProps & {
    /** Add underlay color to the accessory of avatar. */
    underlayColor?: ColorValue;

    /** Add custom styling to the accessory of avatar. */
    style?: StyleProp<ViewStyle>;
  };

/** This is used for adding an accessory to the Avatar.
 * Receives either all [Icon](icon#props) or [Image](image#props) props. */
export const Accessory: RneFunctionComponent<AccessoryProps> = ({
  size = 10,
  style,
  underlayColor = '#000',
  onPress,
  onLongPress,
  onPressIn,
  onPressOut,
  source,
  pressableProps,
  ...rest
}: AccessoryProps) => {
  return (
    <Pressable
      {...{
        android_ripple:
          (onPress || onLongPress) && androidRipple(underlayColor),
        ...pressableProps,
      }}
      style={[
        styles.accessory,
        {
          width: size,
          height: size,
          borderRadius: size / 2,
        },
        style,
      ]}
      {...{ onPressOut, onPressIn, onPress, onLongPress }}
    >
      <View>
        {source ? (
          <Image
            source={source}
            style={{
              width: size,
              height: size,
              borderRadius: size / 2,
            }}
            {...rest}
          />
        ) : (
          <Icon
            name="mode-edit"
            type="material"
            color="#fff"
            size={size * 0.8}
            {...rest}
          />
        )}
      </View>
    </Pressable>
  );
};

const styles = StyleSheet.create({
  accessory: {
    position: 'absolute',
    bottom: 0,
    right: 0,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#aaa',
    ...Platform.select({
      android: {
        elevation: 1,
      },
      default: {
        shadowColor: '#000',
        shadowOffset: { width: 1, height: 1 },
        shadowRadius: 2,
        shadowOpacity: 0.5,
      },
    }),
  },
});

Accessory.displayName = 'Avatar.Accessory';
