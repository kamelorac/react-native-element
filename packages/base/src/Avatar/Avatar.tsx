import React from 'react';
import {
  View,
  Text,
  Image as RNImage,
  StyleSheet,
  Pressable,
  StyleProp,
  ViewStyle,
  TextStyle,
  ImageSourcePropType,
  ImageURISource,
  ImageStyle,
} from 'react-native';
import {
  InlinePressableProps,
  renderNode,
  RneFunctionComponent,
} from '../helpers';
import { IconObject, Icon } from '../Icon';
import { ImageProps, Image } from '../Image';

export const avatarSizes = {
  small: 34,
  medium: 50,
  large: 75,
  xlarge: 150,
};

type AvatarIcon = IconObject & {
  iconStyle?: StyleProp<TextStyle>;
};

export interface AvatarProps extends InlinePressableProps {
  /** Component for enclosing element (eg: TouchableHighlight, View, etc).
   *
   *  @default `Press handlers present then Pressable else View`
   */
  Component?: typeof React.Component;

  /** Callback function when pressing component. */
  onPress?(): void;

  /** Callback function when long pressing component. */
  onLongPress?(): void;

  /** Styling for outer container. */
  containerStyle?: StyleProp<ViewStyle>;

  /** Image source to be displayed on avatar. */
  source?: ImageSourcePropType;

  /** Style for avatar image. */
  avatarStyle?: ImageStyle;

  /** Makes the avatar circular. */
  rounded?: boolean;

  /** Renders title in the placeholder. */
  title?: string;

  /** Style for the title. */
  titleStyle?: StyleProp<TextStyle>;

  /** Style for the view outside image or icon. */
  overlayContainerStyle?: StyleProp<TextStyle>;

  /** Displays an icon as the main content of the Avatar. **Cannot be used alongside title**. When used with the `source` prop it will be used as the placeholder. */
  icon?: AvatarIcon;

  /** Extra styling for icon component. */
  iconStyle?: StyleProp<TextStyle>;

  /** Size of the avatar. */
  size?: ('small' | 'medium' | 'large' | 'xlarge') | number;

  /** Adds style to the placeholder wrapper. */
  placeholderStyle?: StyleProp<ViewStyle>;

  /** Custom placeholder element (by default, it's the title). */
  renderPlaceholderContent?: React.ReactElement<{}>;

  /** Optional properties to pass to the avatar e.g "resizeMode". */
  imageProps?: Partial<ImageProps>;

  /** Custom ImageComponent for Avatar. */
  ImageComponent?: React.ComponentClass;
}

/**
 * Avatars are found all over ui design from lists to profile screens.
 * They are commonly used to represent a user and can contain photos, icons, or even text.

```
 * */
export const Avatar: RneFunctionComponent<AvatarProps> = ({
  onPress,
  onLongPress,
  onPressIn,
  onPressOut,
  Component = onPress || onLongPress || onPressIn || onPressOut
    ? Pressable
    : View,
  containerStyle,
  icon,
  iconStyle,
  source,
  size = 'small',
  avatarStyle,
  rounded,
  title,
  titleStyle,
  overlayContainerStyle,
  imageProps,
  placeholderStyle,
  renderPlaceholderContent,
  ImageComponent = RNImage,
  children,
  pressableProps,
  ...rest
}) => {
  const width =
    typeof size === 'number' ? size : avatarSizes[size] || avatarSizes.small;

  const height = width;
  const titleSize = width / 2;
  const iconSize = width / 2;

  const PlaceholderContent =
    (renderPlaceholderContent &&
      renderNode(undefined, renderPlaceholderContent)) ||
    (title && (
      <Text
        style={StyleSheet.flatten([
          styles.title,
          { fontSize: titleSize },
          titleStyle,
        ])}
      >
        {title}
      </Text>
    )) ||
    (icon && (
      <Icon
        style={StyleSheet.flatten([iconStyle && iconStyle])}
        color={icon.color || 'white'}
        name={icon.name || 'account'}
        size={icon.size || iconSize}
        type={icon.type || 'material-community'}
      />
    ));

  const hidePlaceholder = !(source && (source as ImageURISource).uri);

  const imageContainerStyle = StyleSheet.flatten([
    styles.overlayContainer,
    rounded && { borderRadius: width / 2, overflow: 'hidden' },
    overlayContainerStyle,
    imageProps && imageProps.containerStyle,
  ]);
  if (imageProps && imageProps.containerStyle) {
    delete imageProps.containerStyle;
  }

  return (
    <Component
      style={StyleSheet.flatten([
        styles.container,
        { height, width },
        rounded && { borderRadius: width / 2 },
        containerStyle,
      ])}
      {...{
        onPress,
        onLongPress,
        onPressIn,
        onPressOut,
        ...pressableProps,
        ...rest,
      }}
    >
      <Image
        testID="RNE__Avatar__Image"
        placeholderStyle={StyleSheet.flatten([
          placeholderStyle,
          hidePlaceholder && styles.hiddenPlaceholderStyle,
        ])}
        PlaceholderContent={PlaceholderContent}
        containerStyle={imageContainerStyle as StyleProp<TextStyle>}
        source={source}
        borderRadius={rounded ? width / 2 : undefined}
        {...imageProps}
        style={StyleSheet.flatten([
          styles.avatar,
          imageProps && imageProps.style,
          avatarStyle,
        ])}
        ImageComponent={ImageComponent}
      />
      {children}
    </Component>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'transparent',
  },
  avatar: {
    flex: 1,
    width: undefined,
    height: undefined,
  },
  overlayContainer: {
    flex: 1,
  },
  title: {
    color: '#ffffff',
    backgroundColor: 'transparent',
    textAlign: 'center',
  },
  hiddenPlaceholderStyle: {
    backgroundColor: 'transparent',
  },
});

Avatar.displayName = 'Avatar';
