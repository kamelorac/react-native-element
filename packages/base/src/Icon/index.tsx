import { Icon, IconProps, IconNode, IconObject, IconType } from './Icon';

export { Icon };
export type { IconProps, IconNode, IconObject, IconType };
