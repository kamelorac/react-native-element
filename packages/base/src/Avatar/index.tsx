import { Avatar as AvatarBase, AvatarProps } from './Avatar';
import { Accessory, AccessoryProps } from './Avatar.Accessory';

const Avatar = Object.assign(AvatarBase, {
  Accessory,
});
export { Avatar, Accessory };

export type { AccessoryProps, AvatarProps };
