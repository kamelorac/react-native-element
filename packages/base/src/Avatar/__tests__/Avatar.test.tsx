import React from 'react';
import { Avatar } from '..';
import { Pressable, View, Image, Text } from 'react-native';
import { renderWithWrapper } from '../../../.ci/testHelper';
import { avatarSizes } from './../Avatar';
import { Icon } from '../../Icon';

describe('Avatar Component', () => {
  jest.useFakeTimers();

  it('should match snapshot', () => {
    const { toJSON } = renderWithWrapper(
      <Avatar source={{ uri: 'https://i.imgur.com/0y8Ftya.jpg' }} />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('should renders rounded', () => {
    const { wrapper } = renderWithWrapper(
      <Avatar source={{ uri: 'https://i.imgur.com/0y8Ftya.jpg' }} rounded />
    );
    expect(wrapper.findByType(View).props.style.borderRadius).toBe(
      avatarSizes.small / 2
    );
  });

  it('should allows custom imageProps', () => {
    const { wrapper } = renderWithWrapper(
      <Avatar
        source={{ uri: 'https://i.imgur.com/0y8Ftya.jpg' }}
        imageProps={{ resizeMode: 'contain' }}
      />
    );
    expect(wrapper.findByType(Image).props.resizeMode).toBe('contain');
  });

  it('should renders touchable if onPress given', () => {
    const { wrapper } = renderWithWrapper(
      <Avatar
        source={{ uri: 'https://i.imgur.com/0y8Ftya.jpg' }}
        onPress={() => null}
      />
    );
    expect(wrapper.findByType(Pressable)).toBeTruthy();
  });

  describe('Sizes', () => {
    it.each`
      size
      ${'small'}
      ${'medium'}
      ${'large'}
      ${'xlarge'}
    `('accepts $size', ({ size }) => {
      const { wrapper } = renderWithWrapper(
        <Avatar
          source={{ uri: 'https://i.imgur.com/0y8Ftya.jpg' }}
          size={size}
        />
      );
      expect(wrapper.findByType(View).props.style).toMatchObject({
        height: avatarSizes[size],
        width: avatarSizes[size],
      });
    });

    it('should accepts a number as size', () => {
      const { wrapper } = renderWithWrapper(
        <Avatar source={{ uri: 'https://i.imgur.com/0y8Ftya.jpg' }} size={30} />
      );
      expect(wrapper.findByType(View).props.style).toMatchObject({
        height: 30,
        width: 30,
      });
    });

    it('should have default size', () => {
      const { wrapper } = renderWithWrapper(
        <Avatar source={{ uri: 'https://i.imgur.com/0y8Ftya.jpg' }} />
      );
      expect(wrapper.findByType(View).props.style).toMatchObject({
        height: avatarSizes.small,
        width: avatarSizes.small,
      });
    });
  });

  describe('Placeholders', () => {
    it('should renders title if given', () => {
      const { wrapper } = renderWithWrapper(
        <Avatar
          source={{ uri: 'https://i.imgur.com/0y8Ftya.jpg' }}
          title="John"
        />
      );
      expect(wrapper.findByType(Text).props.children).toBe('John');
    });
    it('should renders custom renderPlaceholderContent', () => {
      const { wrapper } = renderWithWrapper(
        <Avatar
          source={{ uri: 'https://i.imgur.com/0y8Ftya.jpg' }}
          renderPlaceholderContent={
            <Text testID="my-custom-placeholder">Hey</Text>
          }
        />,
        'my-custom-placeholder'
      );
      expect(wrapper.props.children).toBe('Hey');
    });

    it('should renders using icon prop', () => {
      const { wrapper } = renderWithWrapper(
        <Avatar
          source={{ uri: 'https://i.imgur.com/0y8Ftya.jpg' }}
          icon={{
            name: 'home',
            type: 'material-community',
          }}
          iconStyle={{
            backgroundColor: 'red',
          }}
        />
      );
      expect(wrapper.findByType(Icon).props).toMatchObject({
        name: 'home',
        type: 'material-community',
        style: { backgroundColor: 'red' },
      });
    });

    it('should render using icon with defaults', () => {
      const { wrapper } = renderWithWrapper(
        <Avatar
          source={{ uri: 'https://i.imgur.com/0y8Ftya.jpg' }}
          iconStyle={{
            backgroundColor: 'red',
          }}
          icon={{}}
        />
      );
      expect(wrapper.findByType(Icon).props).toMatchObject({
        name: 'account',
        color: 'white',
        style: { backgroundColor: 'red' },
      });
    });

    it('should render using image with imageProps', () => {
      const { wrapper } = renderWithWrapper(
        <Avatar
          source={{ uri: 'https://i.imgur.com/0y8Ftya.jpg' }}
          imageProps={{ style: { backgroundColor: 'red' } }}
        />
      );
      const img = wrapper.findByType(Image);
      expect(img.props.source).toMatchObject({
        uri: 'https://i.imgur.com/0y8Ftya.jpg',
      });
      expect(img.props.style).toMatchObject({
        backgroundColor: 'red',
      });
    });

    it('should not show placeholder if not using source', () => {
      const { wrapper } = renderWithWrapper(
        <Avatar
          size="medium"
          overlayContainerStyle={{ backgroundColor: 'pink' }}
          title="MD"
        />,
        'RNE__Image__placeholder'
      );
      expect(wrapper.props.style.backgroundColor).toBe('transparent');
    });

    it("shouldn't show placeholder if source doesn't contain uri property", () => {
      const { wrapper } = renderWithWrapper(
        <Avatar
          size="medium"
          overlayContainerStyle={{ backgroundColor: 'pink' }}
          title="MD"
          source={{}}
        />,
        'RNE__Image__placeholder'
      );
      expect(wrapper.props.style.backgroundColor).toBe('transparent');
    });

    it("shouldn't show placeholder if source exists but uri is undefined", () => {
      const { wrapper } = renderWithWrapper(
        <Avatar
          size="medium"
          overlayContainerStyle={{ backgroundColor: 'pink' }}
          title="MD"
          source={{ uri: undefined }}
        />,
        'RNE__Image__placeholder'
      );
      expect(wrapper.props.style.backgroundColor).toBe('transparent');
    });
  });
});
