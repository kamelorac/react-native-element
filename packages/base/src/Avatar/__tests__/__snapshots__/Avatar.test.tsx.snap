// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <View
    style={
      {
        "backgroundColor": "transparent",
        "height": 34,
        "width": 34,
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        {
          "backgroundColor": "transparent",
          "flex": 1,
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <Image
        onLoad={[Function]}
        source={
          {
            "uri": "https://i.imgur.com/0y8Ftya.jpg",
          }
        }
        style={
          {
            "bottom": 0,
            "flex": 1,
            "height": undefined,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": undefined,
          }
        }
        testID="RNE__Avatar__Image"
        transitionDuration={360}
      />
      <View
        accessibilityElementsHidden={true}
        collapsable={false}
        importantForAccessibility="no-hide-descendants"
        pointerEvents="none"
        style={
          {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#bdbdbd",
              "flex": 1,
              "height": undefined,
              "justifyContent": "center",
              "width": undefined,
            }
          }
          testID="RNE__Image__placeholder"
        />
      </View>
      <View
        style={
          {
            "flex": 1,
            "height": undefined,
            "width": undefined,
          }
        }
        testID="RNE__Image__children__container"
      />
    </View>
  </View>
</View>
`;
